studio/src/queries/collections.ts

```diff
@@ -0,0 +1,287 @@
+      fetch(`/v0/collections/${collectionId}/items/${itemId}`, {
+        method: "DELETE",
+      }).then(async (r) => {
+        if (!r.ok) {
+          const result = await responseToJsonOrText(r);
+          throw resultToError(result);
+        }
+
+        return undefined;
+      }),
+    onSuccess: () => {
+      queryClient.invalidateQueries({ queryKey: [COLLECTIONS_KEY] });
+    },
+    mutationKey: [collectionId, COLLECTIONS_KEY],
+  });
+
+  return mutation;
+}
+
+export function useDeleteCollection(collectionId: number) {
+  const queryClient = useQueryClient();
+  const mutation = useMutation({
+    mutationFn: () =>
+      fetch(`/v0/collections/${collectionId.toString()}`, {
+        method: "DELETE",
+      }).then(async (r) => {
+        if (!r.ok) {
+          const result = await responseToJsonOrText(r);
+          throw resultToError(result);
+        }
+
+        return undefined;
+      }),
+    onSuccess: () => {
+      queryClient.invalidateQueries({ queryKey: [COLLECTIONS_KEY] });
+    },
+    mutationKey: [collectionId, COLLECTIONS_KEY],
+  });
+
+  return mutation;
+}
+
+export function useUpdateCollectionItem() {
+  const queryClient = useQueryClient();
+  const mutation = useMutation({
+    mutationFn: ({
+      collectionId,
+      itemId,
+      extraParams,
+    }: {
+      collectionId: number;
+      itemId: number;
+      extraParams: CollectionItemParams;
+    }) => {
+      const overrideValues = Object.fromEntries(
+        Object.entries(extraParams).filter(([_, value]) => value !== null),
+      ) as CollectionItemParams;
+
+      const state = getStudioStoreState();
+      const collection = state.collections.find((c) => c.id === collectionId);
+      const collectionItem = collection?.collectionItems.find(
+        (i) => i.id === itemId,
+      );
+      const params: CollectionItemParams = {
+        name: collectionItem?.name ?? undefined,
+        requestBody: collectionItem?.requestBody ?? undefined,
+        requestHeaders: collectionItem?.requestHeaders ?? undefined,
+        requestQueryParams: collectionItem?.requestQueryParams ?? undefined,
+        requestPathParams: collectionItem?.requestPathParams ?? undefined,
+        ...overrideValues,
+      };
+      return fetch(
+        `/v0/collections/${collectionId.toString()}/items/${itemId}`,
+        {
+          method: "PUT",
+          headers: {
+            "Content-Type": "application/json",
+          },
+          body: JSON.stringify(params),
+        },
+      ).then(async (r) => {
+        if (!r.ok) {
+          const result = await responseToJsonOrText(r);
+          throw resultToError(result);
+        }
+
+        return undefined;
+      });
+    },
+    onSuccess: () => {
+      queryClient.invalidateQueries({ queryKey: [COLLECTIONS_KEY] });
+    },
+    mutationKey: [COLLECTIONS_KEY, "update-collection-item"],
+  });
+
+  return mutation;
+}
+
+async function responseToJsonOrText(r: Response) {
```

Comment:
added this to factor out the ternary that was happening in most `if (!res.ok) { ... }` blocks